### PR TITLE
extension feature complete

### PR DESCRIPTION
**Pull request by faga295**
feature
* dynamic update
* scroll sync
* snippets
* local image preview

### CODE REVIEW

File: extensions/.yarnrc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+registry "https://registry.npm.taobao.org"
```
Comment (diauweb) [inline extensions/.yarnrc]: 考虑在用户目录而不是repo内设置yarnrc 切换registry可能会减慢ci的速度

**Comment by FangSen9000** [comment]
come on！

File: extensions/remark-renderer/preview-src/index.ts [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                vscode.postMessage({
+                    command:'revealLine',
+                    line
+                });
+            });
+        });
+    }
+    
+    public getCodeLineElements(){
+    for(const element of Array.from(document.getElementsByClassName(this.codeLineClass))){
```
Comment (FangSen9000) [inline extensions/remark-renderer/preview-src/index.ts]: Looks like you need to pay attention to the indentation

File: extensions/remark-renderer/preview-src/index.ts [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+                    root.replaceWith(newContent.querySelector('.md-typeset')!);
+                    // @ts-ignore
+                    MathJax.typeset();
+                    this.getCodeLineElements();
+                    return;
+            }
+        });
+        window.addEventListener('scroll',() => {
+            window.requestAnimationFrame(() =>{
+                if(this.isFromEditor) return;
```
Comment (diauweb) [inline extensions/remark-renderer/preview-src/index.ts:46]: respect eslint rule

**Pull request merged**